File: app/src/main/java/com/capstone/project_niyakneyak/ui/main/etc/AlarmDataView.java
package com.capstone.project_niyakneyak.ui.main.etc;

public class AlarmDataView {
    private boolean sunday, monday, tuesday, wednesday, thursday, friday, saturday;

    public AlarmDataView(){}
    public AlarmDataView(boolean sunday, boolean monday, boolean tuesday, boolean wednesday, boolean thursday, boolean friday, boolean saturday) {
        this.sunday = sunday;
        this.monday = monday;
        this.tuesday = tuesday;
        this.wednesday = wednesday;
        this.thursday = thursday;
        this.friday = friday;
        this.saturday = saturday;
    }

    public boolean isSunday() {return sunday;}

    public void setSunday(boolean sunday) {
        this.sunday = sunday;
    }

    public boolean isMonday() {
        return monday;
    }

    public void setMonday(boolean monday) {
        this.monday = monday;
    }

    public boolean isTuesday() {
        return tuesday;
    }

    public void setTuesday(boolean tuesday) {
        this.tuesday = tuesday;
    }

    public boolean isWednesday() {
        return wednesday;
    }

    public void setWednesday(boolean wednesday) {
        this.wednesday = wednesday;
    }

    public boolean isThursday() {
        return thursday;
    }

    public void setThursday(boolean thursday) {
        this.thursday = thursday;
    }

    public boolean isFriday() {
        return friday;
    }

    public void setFriday(boolean friday) {
        this.friday = friday;
    }

    public boolean isSaturday() {
        return saturday;
    }

    public void setSaturday(boolean saturday) {
        this.saturday = saturday;
    }

    public String getDisplayData(){
        StringBuilder builder = new StringBuilder();
        for(int i = 0; i < 7; i++){
            switch(i){
                case 0->{if(sunday) builder.append("Sun");}
                case 1->{if(monday) {if(!builder.toString().isEmpty()) builder.append(","); builder.append("Mon");}}
                case 2->{if(tuesday) {if(!builder.toString().isEmpty()) builder.append(","); builder.append("Tue");}}
                case 3->{if(wednesday) {if(!builder.toString().isEmpty()) builder.append(","); builder.append("Wed");}}
                case 4->{if(thursday) {if(!builder.toString().isEmpty()) builder.append(","); builder.append("Thu");}}
                case 5->{if(friday) {if(!builder.toString().isEmpty()) builder.append(","); builder.append("Fri");}}
                case 6->{if(saturday) {if(!builder.toString().isEmpty()) builder.append(","); builder.append("Sat");}}
            }
        }
        return builder.toString();
    }
}

File: app/src/main/java/com/capstone/project_niyakneyak/main/etc/AlarmDataView.kt
package com.capstone.project_niyakneyak.main.etc

class AlarmDataView {
    var isSunday = false
    var isMonday = false
    var isTuesday = false
    var isWednesday = false
    var isThursday = false
    var isFriday = false
    var isSaturday = false

    constructor()
    constructor(
        sunday: Boolean,
        monday: Boolean,
        tuesday: Boolean,
        wednesday: Boolean,
        thursday: Boolean,
        friday: Boolean,
        saturday: Boolean
    ) {
        isSunday = sunday
        isMonday = monday
        isTuesday = tuesday
        isWednesday = wednesday
        isThursday = thursday
        isFriday = friday
        isSaturday = saturday
    }

    val displayData: String
        get() {
            val builder = StringBuilder()
            for (i in 0..6) {
                when (i) {
                    0 -> {
                        if (isSunday) builder.append("Sun")
                    }

                    1 -> {
                        if (isMonday) {
                            if (!builder.toString().isEmpty()) builder.append(",")
                            builder.append("Mon")
                        }
                    }

                    2 -> {
                        if (isTuesday) {
                            if (!builder.toString().isEmpty()) builder.append(",")
                            builder.append("Tue")
                        }
                    }

                    3 -> {
                        if (isWednesday) {
                            if (!builder.toString().isEmpty()) builder.append(",")
                            builder.append("Wed")
                        }
                    }

                    4 -> {
                        if (isThursday) {
                            if (!builder.toString().isEmpty()) builder.append(",")
                            builder.append("Thu")
                        }
                    }

                    5 -> {
                        if (isFriday) {
                            if (!builder.toString().isEmpty()) builder.append(",")
                            builder.append("Fri")
                        }
                    }

                    6 -> {
                        if (isSaturday) {
                            if (!builder.toString().isEmpty()) builder.append(",")
                            builder.append("Sat")
                        }
                    }
                }
            }
            return builder.toString()
        }
}